src/util/devices.tsx

```diff
@@ -62,3 +65,26 @@ export const deviceKeyToLabel = (input: string): string => {
   }
   return input;
 };
+
+export const getExistingDeviceNames = (
+  values: InstanceAndProfileFormValues,
+  profiles: LxdProfile[],
+): string[] => {
+  const existingDeviceNames: string[] = [];
+
+  // Returns devices within the instance
+  existingDeviceNames.push(...values.devices.map((item) => item.name));
+  // Returns devices within the profile
+  if (values.entityType === "instance") {
+    const appliedProfiles = getAppliedProfiles(values, profiles);
+    for (const profile of appliedProfiles) {
+      Object.values(profile.devices).forEach((device) => {
+        if ("name" in device) {
+          existingDeviceNames.push((device as { name: string }).name);
+        }
+      });
```

Comment:
The device.name is actually not the right thing to use here, we need the keys of the devices. This might make this block simpler.
You can test by applying a profile with any device i.e. a GPU in it. Attach the profile to an instance. Then add another GPU directly to that instance, the name chosen should be gpu-2, because gpu-1 is taken by a device inherited from the profile. This currently doesn't work, the inherited and the directly added GPU have the same name gpu-1